doc/api/child_process.markdown

```diff
@@ -758,13 +511,368 @@ throw.  The [`Error`][] object will contain the entire result from
+and `killSignal` is sent, the method won't return until the process has
+completely exited. Note that if the process intercepts and handles the
+`SIGTERM` signal and doesn't exit, the parent process will wait until the child
 process has exited.
 
+## Class: ChildProcess
+
+Instances of the `ChildProcess` class represent spawned child processes.
+
+`ChildProcess` is an [`EventEmitter`][].
+
+Child processes always have three associated streams: `child.stdin`,
+`child.stdout`, and `child.stderr`.  These might be shared with the stdio
+streams of the parent process, or they may be separate stream objects
+which can be piped to and from.
+
+Instances of `ChildProcess` are not intended to be created directly. Rather,
+Use the [`child_process.spawn()`][], [`child_process.exec()`][],
+[`child_process.execFile()`][], or [`child_process.fork()`][] methods (or
+their synchronous counterparts) to create instances of `ChildProcess`.
+
+### Event: 'close'
+
+* `code` {Number} the exit code, if it exited normally.
+* `signal` {String} the signal passed to kill the child process, if it
+  was killed by the parent.
+
+The `'close'` event is emitted when the stdio streams of a child process have
+all terminated. This is distinct from the `'exit'` event, since multiple
+processes might share the same stdio streams.
+
+### Event: 'disconnect'
+
+The `'disconnect'` event is emitted after calling the `.disconnect()` method in
+the parent or child process. After disconnecting it is no longer possible to
+send or receive messages, and the `.connected` property is false.
+
+### Event:  'error'
+
+* `err` {Error Object} the error.
+
+The `'error'` event is emitted whenever:
+
+1. The process could not be spawned, or
+2. The process could not be killed, or
+3. Sending a message to the child process failed.
+
+Note that the `'exit'` event may or may not fire after an error has occurred.
+If you are listening to both the `'exit'` and `'error'` events, it is important
+to guard against accidentally invoking handler functions multiple times.
+
+See also [`ChildProcess#kill()`][] and [`ChildProcess#send()`][].
+
+### Event:  'exit'
+
+* `code` {Number} the exit code, if it exited normally.
+* `signal` {String} the signal passed to kill the child process, if it
+  was killed by the parent.
+
+The `'exit'` event is emitted after the child process ends. If the process
+terminated normally, `code` is the final exit code of the process, otherwise
+`null`. If the process terminated due to receipt of a signal, `signal` is the
+string name of the signal, otherwise `null`.
+
+Note that when the `'exit'` event is triggered, child process stdio streams
+might still be open.
+
+Also, note that Node.js establishes signal handlers for `SIGINT` and
+`SIGTERM` and Node.js processes will not terminate immediately due to receipt
+of those signals. Rather, Node.js will perform a sequence of cleanup actions
+then exit normally.
```

Comment:
I think somewhere along the line (1.x?) node started to kill itself by reraising the signal after cleaning up, so its termination status will be by signal (as it should)... or perhaps I misremember.